contracts/sol6/KyberStorage.sol

```diff
@@ -375,16 +375,23 @@ contract KyberStorage is IKyberStorage, PermissionGroupsNoModifiers {
         reserveIds = reservesPerTokenSrc[token];
     }
 
-    function getReserveAddressesPerTokenSrc(address token)
+    function getReserveAddressesPerTokenSrc(address token, uint256 startIndex, uint256 endIndex)
         external
         view
         override
         returns (address[] memory reserveAddresses)
     {
         bytes32[] memory reserveIds = reservesPerTokenSrc[token];
-        reserveAddresses = new address[](reserveIds.length);
-        for(uint i = 0; i < reserveAddresses.length; i++) {
-            reserveAddresses[i] = reserveIdToAddresses[reserveIds[i]][0];
+        if (reserveIds.length == 0) {
+            return reserveAddresses ;
```

Comment:
can we return without init?
seems its working ok with the tests...
right?

Reply:
Yeah it is working fine. Both in tests and also in deployed contracts

Reply:
Goood